app/assets/javascripts/service_request.coffee

```diff
@@ -0,0 +1,119 @@
+setEquipment = ->
+  equipments = $('.select_location option:selected').data('equipments')
+  $('.select_equipment').find('option').remove()
+  $('.select_equipment').append $('<option></option>')
+  $('.existing-equipment').addClass('hidden')
+  $.each equipments, (i, e) ->
+    dataEquipment = JSON.stringify e
+    $('.select_equipment').append $('<option></option>').attr({'value': e.id, 'data-equipment': dataEquipment }).text(e.id)
+
+$('.select_location').livequery ->
+  setEquipment()
+
+$(document).on 'change', '.select_location', ->
+  setEquipment()
+  
+setEquipmentFields = ->
+  equipment = $('.select_equipment option:selected').data('equipment')
+  $('.new-equipment').addClass('hidden')
+  $('.equipment-warranty').addClass('hidden')
+  if equipment
+    $.ajax
+      url: '/equipment_item'
+      type: 'GET'
+      dataType: 'json'
+      data:
+        location_id: $('.select_location').val()
+        id: $('.select_equipment').val()
+      success: (data) ->
+        $('#service_request_category_id').append("<option selected='selected' value='+data.category.attributes.id+'>'+data.category.attributes.name+'</option>")
```

Comment:
One more fact why I don't like current solution - you have to hardcode HTML in JS functions, which is bad practice.